main.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2020 Sebastian Pipping <sebastian@pipping.org>
+ * Licensed under AGPL v3 or later
+ */
+package main
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/hartwork/go-wait-for-it/cli"
+	"github.com/hartwork/go-wait-for-it/logging"
+	"github.com/hartwork/go-wait-for-it/syntax"
+)
+
+type ConnectResult struct {
+	address  syntax.Address
+	duration time.Duration
+	success  bool
+}
+
+func waitForAddress(address syntax.Address) chan bool {
+	available := make(chan bool)
```

Comment:
Make sure to `close(available)` too. Otherwise the channel stays around and (especially for long lived programs) will become a problem.

Reply:
I'd have to close it in `waitForAddressWithTimeout`. I think I get in trouble with the timeout then: If the timeout hits, I close the channel, the goroutine succeeds and writes to the channel: Wouldn't that fail then and maybe even crash the program?

I out that to the test and it was indeed a problem, at least in my scenario.
Here's my `demo.go `:

```go
package main

import "time"

func writeForever(channel chan int) {
    for {
        channel <- 1
    }
}

func main() {
    channel := make(chan int)
    go writeForever(channel)
    close(channel)
    time.Sleep(10 * time.Second)
}
```

When I run it I get:

```console
# go build . && "./$(basename "$PWD")"
panic: send on closed channel

goroutine 5 [running]:
main.writeForever(0xc000016120)
        /tmp/tmp.pWKklKvfrd/demo.go:7 +0x37
created by main.main
        /tmp/tmp.pWKklKvfrd/demo.go:13 +0x58
```

Any ideas how to not have that problem, in general or in the very case of `go-wait-for-it` here?